app/main.py

```diff
@@ -1,3 +1,84 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    Variables = "Distance", int, float
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+        raise TypeError("Unsupported type for addition")
+
+    def __iadd__(self, other: Variables) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported type for addition")
+
+        return self
+
+    def __mul__(self, other: Variables) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+        raise TypeError("Unsupported type for addition")
+
+    def __truediv__(self, other: Variables) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(round((self.km / other), 2))
+
+        raise TypeError("Unsupported type for addition")
+
+    def __lt__(self, other: Variables) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        elif isinstance(other, (int, float)):
+            return self.km < other
+
+        raise TypeError("Unsupported type for addition")
+
+    def __gt__(self, other: Variables) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        elif isinstance(other, (int, float)):
+            return self.km > other
+
+        raise TypeError("Unsupported type for addition")
+
+    def __eq__(self, other: Variables) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        elif isinstance(other, (int, float)):
+            return self.km == other
+
+        raise TypeError("Unsupported type for addition")
+
+    def __le__(self, other: Variables) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        elif isinstance(other, (int, float)):
+            return self.km <= other
+
+        raise TypeError("Unsupported type for addition")
```

Comment:
The error message 'Unsupported type for addition' is misleading in the `__le__` method. It should be 'Unsupported type for comparison'.